src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -297,6 +298,25 @@ class AvroSuite extends FunSuite with BeforeAndAfterAll {
     }
   }
 
+  test("Date field type") {
+    TestUtils.withTempDir { dir =>
+      val schema = StructType(Seq(
+        StructField("float", FloatType, true),
+        StructField("date", DateType, true)
+      ))
+      val rdd = spark.sparkContext.parallelize(Seq(
+        Row(1f, null),
+        Row(2f, new Date(1451980800000L)),
+        Row(3f, new Date(1460098800000L))
+      ))
+      val df = spark.createDataFrame(rdd, schema)
+      df.write.avro(dir.toString)
+      assert(spark.read.avro(dir.toString).count == rdd.count)
+      assert(spark.read.avro(dir.toString).select("date").collect().map(_(0)).toSet ==
+        Array(null, 1451980800000L, 1460098800000L).toSet)
```

Comment:
I'm not familiar with java.sql.Date, but it seemed to be tossing away more than just miliseconds - e.g. when I added seconds part to these timestamps, the test no longer worked. Not sure why ...